### PR TITLE
Feature/cuong create layout

### PR DESCRIPTION
**Pull request by ghost**
#3 Modified footer layout

### CODE REVIEW

**Comment by at-anhnguyen2** [comment]
Please using `font-awesome.css`.

**Review by at-anhnguyen2 — CHANGES_REQUESTED** [review]
Please check my comment!

File: src/app/features/schedule/schedule.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<div class="schedule">
+  <section *ngFor="let item of schedules">
+    <header class="feature-title">
```
Comment (at-anhnguyen2) [inline src/app/features/schedule/schedule.component.html]: No need `header`.

File: src/app/features/schedule/schedule.component.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      <h4>Group {{item.group}}</h4>
+    </header>
+    <table>
```
Comment (at-anhnguyen2) [inline src/app/features/schedule/schedule.component.html]: Please classname!

File: src/app/features/schedule/schedule.component.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
+table {
+  width: 100%;
+}
+.B {
```
Comment (at-anhnguyen2) [inline src/app/features/schedule/schedule.component.scss]: Convert to `snake-case`.

File: src/app/features/schedule/schedule.component.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+table {
```
Comment (at-anhnguyen2) [inline src/app/features/schedule/schedule.component.scss]: If you want using `tag` to stylesheet, please define it on `_base.scss` or `_common.scss`.

File: src/app/features/schedule/schedule.component.scss [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+.B {
+  width: 20px;
+}
+th, td {
+  border: 1px solid black;
+  text-align: center;
+  padding: 10px;
+}
+
+thead {color:green;}
```
Comment (at-anhnguyen2) [inline src/app/features/schedule/schedule.component.scss]: Please define classname for color text.
Ex: `.txt-green`

File: src/app/features/team-registration/team-registration.component.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<div class="container">
+  <form #eleForm="ngForm" (ngSubmit)="submit(eleForm)">
+    <div class="row">
+      <div class="col-md-12 overwirte">
+        <label for="">Name: </label>
```
Comment (at-anhnguyen2) [inline src/app/features/team-registration/team-registration.component.html]: for what?

File: src/app/features/team-registration/team-registration.component.html [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-md-12 overwirte cover">
+        <label for="">Cover: </label>
+        <img class="cover" [src]="imageCover" alt="">
+        <input type="file" name="cover" [(ngModel)]="cover" (change)="handleFileInput2($event.target.files, eleForm)">
+      </div>
+    </div>
+    <button [mat-dialog-close]="team">Add Team</button>
```
Comment (at-anhnguyen2) [inline src/app/features/team-registration/team-registration.component.html]: Don't use `mat` ui.

File: src/app/features/team-registration/team-registration.component.scss [hunk 8]
```diff
@@ -0,0 +1,78 @@
+// .row {
```
Comment (at-anhnguyen2) [inline src/app/features/team-registration/team-registration.component.scss]: Please remove comment code!

File: src/app/features/team-registration/team-registration.component.scss [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    width: 200px;
+    height: 150px;
+  }
+  input {
+    border: 0;
+    width: 64%;
+  }
+}
+
+label {
```
Comment (at-anhnguyen2) [inline src/app/features/team-registration/team-registration.component.scss]: Please define common style for Form!

**Review by at-anhnguyen2 — CHANGES_REQUESTED** [review]

**Comment by at-anhnguyen2** [comment]
- Don't use library!
- Remove .spec.ts file
- Wrong css position

File: .gitignore [hunk 10]
```diff
@@ -5,6 +5,30 @@
...
+/src/app/shared/services/api.service.spec.ts
+/src/app/shared/services/auth.service.spec.ts
+/src/app/shared/services/dialog.service.spec.ts
+/src/app/shared/services/interaction.service.spec.ts
+/src/app/shared/services/match.service.spec.ts
+/src/app/shared/services/schedule.service.spec.ts
+/src/app/shared/services/transfere.service.spec.ts
+
+/src/app/app.component.spec.ts
+
```
Comment (at-anhnguyen2) [inline .gitignore]: *.spec.ts

File: src/app/features/bracket/bracket.component.html [hunk 11]
```diff
@@ -1,174 +1,238 @@
...
-        <div class="panel-top">
-          Panel content
-        </div>
-        <div class="panel-bottom">
-          Panel content
+<div class="row bracket-wrapper">
+  <p>Knockout stage {{ tournamentName }}</p>
+  <div *ngFor="let match of matches" class="col-12 auto-justify">
+    <div class="row">
+      <div *ngIf="match && match.id === 2.1" class="col-1">
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.html]: `match?id === 2.1`

File: src/app/features/bracket/bracket.component.html [hunk 12]
```diff
@@ -1,174 +1,238 @@
...
       </div>
-    </div>
-    <div class="col-3">
-      <div class="panel panel-default content-child">
-        <div class="panel-top">
-          Panel content
-        </div>
-        <div class="panel-bottom">
-          Panel content
+      <div *ngIf="match && match.id !== 2.1" class="col-1">
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.html]: Too many duplicate!

File: src/scss/base/_variables.scss [hunk 13]
```diff
@@ -20,6 +20,7 @@ $color-grey: #808080;
 $color-primary: #62c6f8;
 $color-primary: #fff;
 $color-second: #000;
+$active-dot:#00afd8;
```
Comment (at-anhnguyen2) [inline src/scss/base/_variables.scss]: Space after `:`

File: src/scss/pages/_bracket.scss [hunk 14]
```diff
@@ -1,69 +1,117 @@
-.bracket-page {
-  padding: 20px;
-  width: 60%;
-  .row-relative {
+.bracket-wrapper {
+  width: 100%;
+  margin-right: 0px;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_bracket.scss]: `0` instead `0px`

File: src/scss/pages/_bracket.scss [hunk 15]
```diff
@@ -1,69 +1,117 @@
...
+    img, span {
+      float: none;
     }
   }
-  .content-child {
+  .col-1 {
+    width: 14%;
+    float: left;
+    padding-right: 15px;
+    padding-left: 15px;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_bracket.scss]: Using short hand padding

File: src/scss/pages/_bracket.scss [hunk 16]
```diff
@@ -1,69 +1,117 @@
...
+  }
+  .col-offset-4 {
+    margin-left: 23.333333%;
+  }
+  .col-offset-5 {
+    margin-left: 27.333333%;
+  }
+  .col-offset-6 {
+    margin-left: 28.333333%;
     margin-top: 40px;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_bracket.scss]: Just only margin-left for offset.
No any margin.

File: src/scss/pages/_bracket.scss [hunk 17]
```diff
@@ -1,69 +1,117 @@
...
+  }
+  .panel-center {
+    height: 84px;
+    img {
+      position: relative;
+      top: -150px;
+      width: 100%;
+    }
+    .panel-heading, .panel-footer {
+      max-width: 174.91px;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_bracket.scss]: Why 174.91?

File: src/scss/pages/_register.scss [hunk 18]
```diff
@@ -0,0 +1,19 @@
+.card-register {
+    min-height: 614px;
+    background-repeat: no-repeat;
+    background-repeat: no-repeat;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_register.scss]: Duplicate!

File: src/scss/pages/_register.scss [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    background-image: linear-gradient(rgb(104, 145, 162), rgb(12, 97, 33));
+    .form-register {
+      max-width: 350px;
+      border-radius: 10px;
+      padding: 40px;
+      margin: 0 auto;
+      transform: translateY(50%);
+      background-color: #F7F7F7;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_register.scss]: `#f7f7f7` should be created variable.

File: src/scss/pages/_team-registration.scss [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+  display: inline-block;
+  width: 10%;
+  margin-bottom: 7px;
+  margin-top: 20px;
+}
+input {
+  height: 36px;
+  padding: 10px;
+  border-radius: 4px;
+  border: 1px solid #e1e1e1;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_team-registration.scss]: `#e1e1e1` should be created variable.

File: src/scss/pages/_team-registration.scss [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+  width: 84%;
+}
+button {
+  margin: 10px 0;
+  padding: 10px 20px;
+  border: 0;
+  border-radius: 4px;
+  background: #000;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_team-registration.scss]: `#000` should be created variable.

File: src/scss/pages/_team-registration.scss [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+  color: #fff;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_team-registration.scss]: `#fff` should be created variable.

File: src/scss/pages/_team-registration.scss [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+}
+.overwirte {
+  float: none;
```
Comment (at-anhnguyen2) [inline src/scss/pages/_team-registration.scss]: Create common classname for `float: none`

**Review by at-anhnguyen2 — CHANGES_REQUESTED** [review]
Please check my comment!

File: src/app/features/bracket/bracket.component.html (hunk 11 above)
Comment (at-cuongbao) [reply src/app/features/bracket/bracket.component.html]: ok anh.

File: src/app/features/bracket/bracket.component.html [hunk 24]
```diff
@@ -1,36 +1,52 @@
...
+            <div class="panel-heading clearfix">
+              <img src="../../../assets/images/logo-img.png" alt="" width="60" height="60">
+              <span>{{ team.code }}</span>
+              <span>{{ team.score }}</span>
+            </div>
+          </div>
+        </div>
+      </ng-container>
+    </div>
+    <!-- <div class="row">
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.html]: Please remove comment code.

File: src/app/features/bracket/bracket.component.ts [hunk 25]
```diff
@@ -1,34 +1,80 @@
 import { Component, OnInit } from '@angular/core';
 import { MatchService } from 'src/app/shared/services/match.service';
 
+const POSITION = {
+  ck: {
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.ts]:
```
const POSITION = {
  ck: 2,
  bk: 4,
  tk: 8
}
```

File: src/app/features/bracket/bracket.component.ts [hunk 26]
```diff
@@ -1,34 +1,80 @@
...
+          return (value.label === POSITION[key].label && value.position === i);
+        });
+        if (team) {
+          this.bracketView.push({
+            label: POSITION[key].label,
+            position: i,
+            code: team ? team.code : '?',
+            score: team && team.score ? team.score : '?'
+          });
+        } else {
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.ts]: No need `if` `else`.

File: src/app/features/bracket/bracket.component.ts [hunk 27]
```diff
@@ -1,34 +1,80 @@
...
         this.tournamentName = data.tournamentName
-        this.winner = data.winner;
       });
   }
 
   isEmpty(obj) {
-    for(var key in obj) {
-      if(obj.hasOwnProperty(key))
+    for (var key in obj) {
+      if (obj.hasOwnProperty(key))
```
Comment (at-anhnguyen2) [inline src/app/features/bracket/bracket.component.ts]: Open and Close `block scope`

File: src/app/features/login/login.component.html [hunk 28]
```diff
@@ -3,16 +3,24 @@
...
+        <p *ngIf="password.errors.minlength" class="alert alert-danger">Password must be at least 8 characters long!</p>
+      </div>
     </div>
-    <button class="btn btn-primary btn-block">Sign in</button>
-    <a href="" class="forgot-password">Forgot the password?</a>
+    <button class="btn btn-primary btn-block" [disabled]="f.invalid">Sign in</button>
+    <a routerLink="/register" class="forgot-password">Haven't had account yet? Go Register</a>
   </form>
-</div>
+</div>
```
Comment (at-anhnguyen2) [inline src/app/features/login/login.component.html]: Space line!

File: src/app/features/register/register.component.html [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+      <label for="inputPassword" class="">Confirm Password</label>
+      <input type="password" id="inputConfirmPassword" name="confirmPassword" #confirmPassword="ngModel" required
+        ngModel class="form-control">
+      <div *ngIf="password.value !== confirmPassword.value && (confirmPassword.dirty || confirmPassword.touched)">
+        <p class="alert alert-danger">Password does not match the confirmed password!</p>
+      </div>
+    </div>
+    <button class="btn btn-primary btn-block" [disabled]="f.invalid">Register</button>
+  </form>
+</div>
```
Comment (at-anhnguyen2) [inline src/app/features/register/register.component.html]: Space line!

File: src/app/features/tournament-registration/tournament-registration.component.ts [hunk 30]
```diff
@@ -33,17 +27,18 @@ export class TournamentRegistrationComponent implements OnInit {
...
   ngOnInit() {
 
   }
 
   // open diablog
   openDialog() {
-    const dialogRef = this.dialog.open(TeamRegistrationComponent, {
-  
-    });
+    const dialogRef = this.dialog.open(TeamRegistrationComponent, {});
```
Comment (at-anhnguyen2) [inline src/app/features/tournament-registration/tournament-registration.component.ts]: What is `this.dialog`

File: src/app/shared/components/header/header.component.html [hunk 31]
```diff
@@ -49,8 +10,8 @@
         <a routerLink="/match" class="main-link" href="#contact">Match</a>
```
Comment (at-anhnguyen2) [inline src/app/shared/components/header/header.component.html]: Remove `href`

File: src/scss/state/_state.scss [hunk 32]
```diff
@@ -10,3 +10,17 @@
...
+  border: 1px solid transparent;
+}
+.alert-danger {
+  color: $color-light-red;
+  background-color: $color-white;
+  border-color: $color-grey;
+}
+.active {
+  color: $color-blue !important;
+}
```
Comment (at-anhnguyen2) [inline src/scss/state/_state.scss]: Space line!

File: src/scss/pages/_register.scss (hunk 18 above)
Comment (at-anhnguyen2) [reply src/scss/pages/_register.scss]: Please remove!

**Review by at-anhnguyen2 — CHANGES_REQUESTED** [review]
Too many comment code unnecessary!
About CSS, some properties should be created to `common class`, such as: `txt-center`, `pull-left`, `w-100`, ...

**Review by at-anhnguyen2 — APPROVED** [review]

**Pull request merged**